src/Umbraco.Core/Persistence/SqlSyntax/SqlServerSyntaxProvider.cs

```diff
@@ -246,6 +249,41 @@ public override bool DoesTableExist(IDatabase db, string tableName)
             return result > 0;
         }
 
+        public override void WriteLock(IDatabase db, params int[] lockIds)
+        {
+            // soon as we get Database, a transaction is started
+
+            if (db.Transaction.IsolationLevel < IsolationLevel.ReadCommitted)
+                throw new InvalidOperationException("A transaction with minimum ReadCommitted isolation level is required.");
+
+
+            // *not* using a unique 'WHERE IN' query here because the *order* of lockIds is important to avoid deadlocks
+            foreach (var lockId in lockIds)
+            {
+                db.Execute(@"SET LOCK_TIMEOUT 1800;");
+                var i = db.Execute(@"UPDATE umbracoLock WITH (REPEATABLEREAD) SET value = value*-1 WHERE id=@id", new { id = lockId });
```

Comment:
if by accident the value has been set to zero this does not lock - hence the original CASE

Reply:
@zpqrtbnk, is there ways this can happen other than people manipulating the database them self? I wonder why we handle this special case?

Reply:
no, it was "assume garbage in" but not critical - you decide